src/buf/fixed/handle.rs

```diff
@@ -33,8 +32,7 @@ pub(super) struct CheckedOutBuf {
 ///
 pub struct FixedBuf {
     registry: Rc<RefCell<dyn FixedBuffers>>,
-    buf: ManuallyDrop<Vec<u8>>,
-    index: u16,
+    buf: CheckedOutBuf,
```

Comment:
There didn't seem any point in duplicating the fields

Reply:
This is OK with the proposed generalization: there is no longer necessarily a `Vec` managing the buffer's memory, so it's better to just store the raw fields. The manually dropped `Vec` was used for convenient access to the slice.

Reply:
That was all i meant - after the changes in FixedBuf, you'd want all the fields from checkedbuf, so reused them.